Wikipedia/Code/TalkPageViewController.swift

```diff
@@ -854,7 +862,14 @@ extension TalkPageViewController: TalkPageTopicComposeViewControllerDelegate {
             case .failure(let error):
                 DDLogError("Failure publishing topic: \(error)")
                 composeViewController.setupNavigationBar(isPublishing: false)
-                // TODO: Display failure banner on topic compose VC
+                let title = TalkPageLocalizedStrings.newTopicFailedAlertTitle
```

Comment:
Update: it looks like you might want to pass along the `composeViewController` parameter and present the server error on that, to avoid the console error:

```
    fileprivate func showUnexpectedErrorAlert(on viewController: UIViewController) {
        let alert = UIAlertController(title: TalkPageLocalizedStrings.unexpectedErrorAlertTitle, message: TalkPageLocalizedStrings.unexpectedErrorAlertSubtitle, preferredStyle: .alert)
        let action = UIAlertAction(title: CommonStrings.okTitle, style: .default)
        alert.addAction(action)
        viewController.present(alert, animated: true)
    }
```

Call for replies with: ` self.showUnexpectedErrorAlert(on: self)` and topics with `self?.showUnexpectedErrorAlert(on: composeViewController)`

Reply:
Great catch, these are also done!